plugins/priceFeed.go

```diff
@@ -38,18 +38,24 @@ func MakePriceFeed(feedType string, url string) (api.PriceFeed, error) {
 		}
 		tickerAPI := api.TickerAPI(exchange)
 		return newExchangeFeed(url, &tickerAPI, &tradingPair), nil
+	case "sdex":
+		SDEXfeed, e := newSDEXFeed(sdex, url)
+		if e != nil {
+			return nil, fmt.Errorf("unable to create SDEX priceFeed ")
+		}
+		return SDEXfeed, nil
 	}
 	return nil, nil
 }
 
 // MakeFeedPair is the factory method that we expose
-func MakeFeedPair(dataTypeA, dataFeedAUrl, dataTypeB, dataFeedBUrl string) (*api.FeedPair, error) {
-	feedA, e := MakePriceFeed(dataTypeA, dataFeedAUrl)
+func MakeFeedPair(sdex *SDEX, dataTypeA, dataFeedAUrl, dataTypeB, dataFeedBUrl string) (*api.FeedPair, error) {
```

Comment:
we should not take in `sdex` here.

The `priceFeed.go` should already hold a reference to the sdex instance before we ever would want to call MakePriceFeed or MakeFeedPair. However, we're not currently set up to allow for that so you can just keep a global variable in priceFeed.go for now, calling it `_privateSdexHack` which you can set in `trade.go` after constructing sdex.go. Please add a comment on the line where you set it to indicate this is a temporary hack.

The right way to do this would be to make an `objectFactory.go` struct that is stateful, containing the sdex instance and all factory methods should hang off this instance (i.e. all Make* methods)

Reply:
see new comment on instantiating sdex directly here instead of using the global variable approach -- we would still need a temporary global variable for certain parts (network and horizon api) to allow instantiating the sdex object